### PR TITLE
lab exercises for langchain components

### PR DESCRIPTION
**Pull request by sanjanalreddy**
- added exercises for gemini langchain components
- minor markdown fixes

### CODE REVIEW

File: notebooks/vertex_genai/labs/gemini_langchain_components.ipynb [hunk 1]
```diff
@@ -0,0 +1,1573 @@
+{
```
Comment (BenoitDherin) [inline notebooks/vertex_genai/labs/gemini_langchain_components.ipynb:1]: <p>It seems some new line is missing, and some of the mardown is not properly rendered.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/514/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

File: notebooks/vertex_genai/labs/palm_langchain_components.ipynb [hunk 2]
```diff
@@ -0,0 +1,1300 @@
+{
```
Comment (BenoitDherin) [inline notebooks/vertex_genai/labs/palm_langchain_components.ipynb:1]: <p>Same comment here as in the Gemini version: newline + formatting</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/514/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

**Review by BenoitDherin — APPROVED** [review]

File: notebooks/vertex_genai/labs/gemini_langchain_components.ipynb (hunk 1 above)
Comment (sanjanalreddy) [reply notebooks/vertex_genai/labs/gemini_langchain_components.ipynb:1]: <p>Updated</p>

File: notebooks/vertex_genai/labs/palm_langchain_components.ipynb (hunk 2 above)
Comment (sanjanalreddy) [reply notebooks/vertex_genai/labs/palm_langchain_components.ipynb:1]: <p>done</p>

**Pull request merged**